### PR TITLE
Flux

### PR DESCRIPTION
**Pull request by LucasZibell**
## Summary
The app now uses flux with redux

## TrelloCard
https://trello.com/c/jP4hKrqW/23-flux-y-redux

### CODE REVIEW

File: React/my-app/src/redux/accounts/actions.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      if (response.status >= 200 && response.status < 300) {
+        dispatch({
+          type: actions.CHECK_USER_SUCCESS,
+          payload: { token: response.data.access_token }
+        });
+      }
+    } catch (e) {
+      dispatch({
+        type: actions.CHECK_USER_FAILURE,
+        payload: { error: 1 }
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/actions.js]: what is error1 = 1 ?
You should be record the error here.

File: React/my-app/src/redux/accounts/actions.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  NEW_USER_FAILURE: "NEW_USER_FAILURE",
+  CHECK_USER_SUCCESS: "CHECK_USER_SUCCESS",
+  CHECK_USER_FAILURE: "CHECK_USER_FAILURE"
+};
+
+export const verifyUser = body => {
+  return async dispatch => {
+    try {
+      const response = await checkUser(body);
+      if (response.status >= 200 && response.status < 300) {
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/actions.js:17]: what if request fail, but just fail, not crash!
So you never get success, but never get failure too.

File: React/my-app/src/redux/accounts/actions.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        dispatch({
+          type: actions.CHECK_USER_SUCCESS,
+          payload: { token: response.data.access_token }
+        });
+      }
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/actions.js]: if statuscode is wrong, throw a new error here, with the request problem
so you ll catch him, dispatch the failure and handle the error! 😄

File: React/my-app/src/redux/accounts/actions.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    } catch (e) {
+      dispatch({
+        type: actions.CHECK_USER_FAILURE,
+        payload: { error: 1 }
+      });
+    }
+  };
+};
+
+export const registerNewUser = body => {
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/actions.js:34]: same

File: React/my-app/src/redux/accounts/reducer.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+const initialState = {
+  token: null,
+  error: null
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case actions.NEW_USER_SUCCESS:
+      return { ...state, error: action.payload.error };
+      break;
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/reducer.js]: brack sentence is unnecessary

File: React/my-app/src/redux/book/actions.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    dispatch({
+      type: actions.LOADING_GALLERY,
+      payload: { loadingGallery: status }
+    });
+  };
+};
+
+export const getBookList = () => {
+  return async dispatch => {
+    const response = await getBookGallery();
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/book/actions.js]: use the try and catch system

File: React/my-app/src/redux/book/actions.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+export const resetGalleryView = () => {
+  return {
+    type: actions.RESET_GALLERY_VIEW
+  };
+};
+
+export const getBook = id => {
+  return async dispatch => {
+    const response = await getBookInfo(id);
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/book/actions.js]: same on the rest

File: React/my-app/src/redux/book/actions.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+      );
+    } else {
+      gallery = data.filter(
+        book =>
+          book.title.toLowerCase().includes(word.toLowerCase()) ||
+          book.author.toLowerCase().includes(word.toLowerCase())
+      );
+    }
+    dispatch({
+      type: actions.GET_FILTER_GALLERY,
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/book/actions.js]: GET_FILTERED_FALLERY ?

File: React/my-app/src/redux/book/reducer.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  bookInfo: null,
+  loading: true,
+  loadingGallery: true
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case actions.GET_FILTER_GALLERY:
+      return { ...state, gallery: action.payload.gallery };
+      break;
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/book/reducer.js]: remove break

File: React/my-app/src/screens/book_detail/layout.js [hunk 10]
```diff
@@ -19,50 +20,53 @@ import {
...
+  ) : (
+    <div className="mainContainer">
+      <NavBarContainer />
+      <Link className="back" to="/">
+        {VOLVER}
+      </Link>
+      <div className="center">
+        <div className="info">
+          <img className="main_book" src={book.image_url || bigDefaultImg} />
+          <div className="description_box">
```
Comment (IgnacioAbadie) [inline React/my-app/src/screens/book_detail/layout.js]: I think that you can clear a little this code if you create specifics components.
For example, `descriptionBox Component` `suggests Component`.
Create this components into the components file in the same folder
```
youScreen
    -components
             -subComponent1
                    -index.js
    - index.js
    - layout.js
```

File: React/my-app/src/screens/sign_up/index.js [hunk 11]
```diff
@@ -106,12 +107,18 @@ class SignUpContainer extends Component {
...
         }
-      },
-      this.onSuccess,
-      this.onFailure
+      })
     );
   }
 
+  componentWillReceiveProps(newProps) {
+    if (newProps.error) {
```
Comment (IgnacioAbadie) [inline React/my-app/src/screens/sign_up/index.js]: i'd replace it by 
`this.setState(newProps.error ? { emailError: MAIL_IN_USE } : { redirect: true })`

File: React/my-app/src/service/accounts.js [hunk 12]
```diff
@@ -1,21 +1,17 @@
...
-    .post(url, body)
-    .then(
-      response =>
-        response.status >= 200 && response.status < 300
-          ? onSuccess(response)
-          : onFailure()
-    )
-    .catch(e => onFailure());
+const failure = () => {
+  console.log("Request Failed");
```
Comment (IgnacioAbadie) [inline React/my-app/src/service/accounts.js]: reqeust fail should be catched on the actions

**Review by IgnacioAbadie — CHANGES_REQUESTED** [review]

File: React/my-app/src/components/Description_Box/styles.css [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+.other_info {
+  font-size: 16px;
+  margin-bottom: 20px;
+}
+
+.description {
+  line-height: 19px;
+  color: #999999;
```
Comment (IgnacioAbadie) [inline React/my-app/src/components/Description_Box/styles.css]: #999

File: React/my-app/src/components/Description_Box/styles.css [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  margin-bottom: 41px;
+  max-width: 563px;
+}
+
+.rent_button {
+  background: #00adee;
+  color: #fff;
```
Comment (IgnacioAbadie) [inline React/my-app/src/components/Description_Box/styles.css]: #FFF

File: React/my-app/src/redux/accounts/actions.js [hunk 15]
```diff
@@ -16,11 +19,13 @@ export const verifyUser = body => {
           type: actions.CHECK_USER_SUCCESS,
           payload: { token: response.data.access_token }
         });
+      } else {
+        throw INVALID_USER;
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/actions.js]: the standard way to throw and handle exceptions is using the Error object.
so
`throw new Error(message);` is that you have to do

File: React/my-app/src/redux/accounts/actions.js [hunk 16]
```diff
@@ -35,11 +40,13 @@ export const registerNewUser = body => {
           type: actions.NEW_USER_SUCCESS,
           payload: { error: 0 }
```
Comment (IgnacioAbadie) [inline React/my-app/src/redux/accounts/actions.js]: don need thats the payload error on the success. Just modifique this when enter on reducer action

**Review by IgnacioAbadie — CHANGES_REQUESTED** [review]
Just a little change more and go.
Anyway, a observation.
When i told you to extract components like suggest or description box, i meant into components folder INSIDE of the view that is working with that.
You create this components in the components folder that is shared with all the aplication.
For example, a custom Buttom component is a component taht you want to share with all the proyect, but a specific book deatil box you only want to show him on the books detials view.

File: React/my-app/src/components/Description_Box/styles.css (hunk 14 above)
Comment (LucasZibell) [reply React/my-app/src/components/Description_Box/styles.css]: The prettier auto corrects me the #FFF and rewrites it as #fff

Comment (IgnacioAbadie) [reply React/my-app/src/components/Description_Box/styles.css]: okas, but keep in mind that we write on MAYUS by convention

**Review by IgnacioAbadie — APPROVED** [review]

**Pull request merged**